### PR TITLE
Rate Limit metrics

### PR DESCRIPTION
**Pull request by kim-tsao**
### What does this PR do?:

Adds prometheus metrics to count the various git operation calls

### Which issue(s)/story(ies) does this PR fixes:

https://issues.redhat.com/browse/DEVHAS-272

### PR acceptance criteria:

- [ ] Unit/Functional tests

  

- [ ] Documentation 

   

- [ ] Client Impact

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -119,6 +116,17 @@ check_fmt:
...
 
+
+### fmt: Runs go fmt against code.  Borrowed from DWO
+fmt:
+  ifneq ($(shell command -v goimports 2> /dev/null),)
+	find . -not -path '*/\.*' -not -name '*zz_generated*.go' -name '*.go' -exec goimports -w {} \;
+  else
+	  @echo "WARN: goimports is not installed -- formatting using go fmt instead."
+	  @echo "      Please install goimports to ensure file imports are consistent."
+	  go fmt -x ./...
```
Comment (johnmcollier) [inline Makefile:127]: Should we move the `go fmt -x ...` to outside of the if statement?

File: pkg/metrics/metrics.go [hunk 2]
```diff
@@ -40,9 +41,42 @@ var (
...
+
+	ControllerGitRequest = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "controller_git_request",
+			Help: "Number of git operation requests.  Not an SLI metric",
+		},
+		[]string{"controller", "tokenName", "operation"},
+	)
+
+	SecondaryRateLimitCounter = prometheus.NewCounterVec(
```
Comment (johnmcollier) [inline pkg/metrics/metrics.go:53]: Are we doing anything with this and the PrimaryRateLimitCounter counters yet? Or is this just adding the counters for now?

**Review by johnmcollier — COMMENTED** [review]
Looks fine - just a couple questions

File: pkg/metrics/metrics.go (hunk 2 above)
Comment (kim-tsao) [reply pkg/metrics/metrics.go:53]: we're just setting counters for now.  Eventually, they will be used in alerting.

File: Makefile (hunk 1 above)
Comment (kim-tsao) [reply Makefile:127]: wouldn't that be redundant if `goimports` has already been run?

File: pkg/metrics/metrics.go (hunk 2 above)
Comment (johnmcollier) [reply pkg/metrics/metrics.go:53]: Sorry, what I mean is, I don’t see this or PrimaryRateLimitCounter set any where in the code? E.g checking for a secondary rate limit error

File: Makefile (hunk 1 above)
Comment (kim-tsao) [reply Makefile:127]: The [docs](https://pkg.go.dev/golang.org/x/tools/cmd/goimports) say it can also be used to format

**Review by johnmcollier — APPROVED** [review]

**Pull request merged**